pandas/core/indexes/datetimes.py

```diff
@@ -1968,14 +1970,32 @@ def tz_localize(self, tz, ambiguous='raise', errors='raise'):
             .. deprecated:: 0.15.0
                Attempt to infer fall dst-transition hours based on order
 
+
         Returns
         -------
         localized : DatetimeIndex
 
+        Examples
+        --------
+        In the example below, We put the date range from 01 March 2018
+        to 08 March 2018 & convert this to US/Eastern Time zone
+
+        >>> df = pd.date_range('2018-03-01', '2018-03-05')
+        >>> df
+        DatetimeIndex(['2018-03-01', '2018-03-02', '2018-03-03',
+        '2018-03-04', '2018-03-05'],
+        dtype='datetime64[ns]', freq='D')
+        >>> df.tz_localize(tz='US/Eastern')
+        DatetimeIndex(['2018-03-01 00:00:00-05:00',
+        '2018-03-02 00:00:00-05:00', '2018-03-03 00:00:00-05:00',
+        '2018-03-04 00:00:00-05:00', '2018-03-05 00:00:00-05:00'],
```

Comment:
showing the reverse example is also useful, IOW ``.tz_localize(None)``, removes the timezone and makes it tz-naive.

Reply:
@jreback when I add another example it shows error how I add that example in the docstring

Reply:
```
n [11]: pd.date_range('2018-03-01', '2018-03-05').tz_localize('US/Eastern').tz_localize(None)
Out[11]: 
DatetimeIndex(['2018-03-01', '2018-03-02', '2018-03-03', '2018-03-04',
               '2018-03-05'],
              dtype='datetime64[ns]', freq='D')
```

Reply:
@jreback @jorisvandenbossche I made required change.